src/DynamoCore/Models/NodeModel.cs

```diff
@@ -1434,17 +1441,42 @@ protected override void DeserializeCore(XmlElement nodeElement, SaveContext cont
         ///     This property forces all AST nodes that generated from this node
         ///     to be executed, even there is no change in AST nodes.
         /// </summary>
-        public virtual bool ForceReExecuteOfNode
+        [Flags]
+        protected enum ExecutionHints
         {
-            get
-            {
-                return forceReExec;
-            }
-            set
-            {
-                forceReExec = value;
-                RaisePropertyChanged("ForceReExecuteOfNode");
-            }
+            None = 0,
+            Modified = 1,
+            ForceExecute = 3
```

Comment:
Do add these comments to indicate that the two flags do overlap in their values:

```csharp
Modified = 1,    // Marks as modified, but execution is optional if AST is unchanged.
ForceExecute = 3 // Marks as modified, force execution even if AST is unchanged.
```